### PR TITLE
TimePicker moves to minute mode after hour selection

### LINKED ISSUE #17276: TimePicker doesn't switch to minute after selecting hour like the Android native time picker does
In the native Android time picker, it automatically switches to minute after you select an hour. In Flutter material, it seems you have to manually tap on the minute part of the time after choosing an hour in order to select a minute.
Just a minor suggestion—it should be an easy fix.

### PR DESCRIPTION
**Pull request by darrenaustin**
## Description

Adds a feature of the native Android Time Picker to our Material Time Picker. When the user selects an hour, it automatically switches to minute mode. 

This is a merging of two pull requests:
- Code changes from @sdolski https://github.com/flutter/flutter/pull/24677
- Tests from @lucaslcode https://github.com/flutter/flutter/pull/29876

Thanks to both of you for your contributions!

## Related Issues

Fixes #17276, #29600

## Tests

The following tests were added from @lucaslcode:

'tap-select switches from hour to minute'
'drag-select switches from hour to minute'

to time_picker_test.dart.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/material/time_picker.dart [hunk 1]
```diff
@@ -1000,12 +1000,14 @@ class _Dial extends StatefulWidget {
     @required this.mode,
     @required this.use24HourDials,
     @required this.onChanged,
+    @required this.onHourSelected,
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/time_picker.dart]: Even though this is private etc, It looks like the implementation assumes that onHourSelected is non-null; should assert as much here. If there similar assumptions about any of the other parameters, maybe more asserts too.

File: packages/flutter/test/material/time_picker_test.dart [hunk 2]
```diff
@@ -134,6 +134,36 @@ void _tests() {
     expect(result.hour, equals(9));
   });
 
+  testWidgets('tap-select switches from hour to minute', (WidgetTester tester) async {
+    TimeOfDay result;
+
+    final Offset center = await startPicker(tester, (TimeOfDay time) { result = time; });
+    await tester.tapAt(Offset(center.dx, center.dy + 50.0));
```
Comment (HansMuller) [inline packages/flutter/test/material/time_picker_test.dart]: It would be nice to define variables for the offsets, like hour6, like in the other tests.

**Review by HansMuller — APPROVED** [review]
LGTM with some trivial changes

**Comment by up3** [comment]
How do I set it to no show the minutes (like t was before) ?

**Pull request merged**